src/dmd/globals.d

```diff
@@ -541,14 +550,28 @@ nothrow:
         }
         if (linnum)
         {
-            buf.writeByte('(');
-            buf.print(linnum);
-            if (showColumns && charnum)
+            final switch (messageStyle)
             {
-                buf.writeByte(',');
-                buf.print(charnum);
+                case MessageStyle.digitalmars:
+                    buf.writeByte('(');
+                    buf.print(linnum);
+                    if (showColumns && charnum)
+                    {
+                        buf.writeByte(',');
+                        buf.print(charnum);
+                    }
+                    buf.writeByte(')');
+                    break;
+                case MessageStyle.gnu: // https://www.gnu.org/prep/standards/html_node/Errors.html
```

Comment:
You could also do:
```D
static immutable GNUStyle = [ ':', ':', '\0' ];
static immutable DMStyle = [ '(', ',', ')' ];
immutable style = messageStyle == MessageStyle.digitalmars ? DMStyle : GNUStyle; 
```

If you want to avoid the code duplication ?

Reply:
I prefer the code duplication. The list seems to imply that one may add a third style, and I'm not sure it's likely that a hypothetical third style would fall into the same format schema.

Reply:
I doubt there will ever be one TBH, but the three entries correspond to "before line", "before column (if any)", and "after line". But it was just a suggestion, fine by me if you prefer this style.